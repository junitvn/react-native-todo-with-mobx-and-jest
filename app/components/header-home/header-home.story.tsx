import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { color } from "../../theme"
import { HeaderHome } from "./header-home"

storiesOf("HeaderHome", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <HeaderHome style={{ backgroundColor: color.error }} />
      </UseCase>
    </Story>
  ))
