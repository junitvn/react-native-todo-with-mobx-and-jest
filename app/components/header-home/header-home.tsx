import * as React from "react"
import { ImageStyle, StyleProp, View, ViewStyle } from "react-native"
import { observer } from "mobx-react-lite"
import { Icon } from "../icon/icon"
import { Gap } from "../gap/gap"
import { spacing } from "../../theme"

const MENU_ICON_SIZE = 40
const NOTI_ICON_SIZE = 25
const SEARCH_ICON_SIZE = 23

const CONTAINER: ViewStyle = {
  justifyContent: "space-between",
  flexDirection: 'row'
}


const MENU_ICON_CONTAINER: ImageStyle = {
  width: MENU_ICON_SIZE,
  height: MENU_ICON_SIZE,
  marginLeft: spacing[2]
}

const NOTI_ICON_CONTAINER: ImageStyle = {
  width: NOTI_ICON_SIZE,
  height: NOTI_ICON_SIZE,
}

const SEARCH_ICON_CONTAINER: ImageStyle = {
  width: SEARCH_ICON_SIZE,
  height: SEARCH_ICON_SIZE,
}

const HEADER_RIGHT: ViewStyle = {
  flexDirection: 'row',
  marginRight: spacing[3]
}

export interface HeaderHomeProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
}

/**
 * Describe your component here
 */
export const HeaderHome = observer(function HeaderHome(props: HeaderHomeProps) {
  const { style } = props
  const styles = Object.assign({}, CONTAINER, style)

  return (
    <View style={styles}>
      <Icon icon="menu" style={MENU_ICON_CONTAINER} />
      <View style={HEADER_RIGHT}>
        <Icon icon="search" style={SEARCH_ICON_CONTAINER} />
        <Gap width={16} />
        <Icon icon="noti" style={NOTI_ICON_CONTAINER} />
      </View>
    </View>
  )
})
