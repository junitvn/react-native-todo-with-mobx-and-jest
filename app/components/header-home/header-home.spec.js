import React from "react"
import renderer from "react-test-renderer"
import { HeaderHome } from "./header-home"

test("renders correctly", () => {
  const tree = renderer.create(<HeaderHome />).toJSON()
  expect(tree).toMatchSnapshot()
})
