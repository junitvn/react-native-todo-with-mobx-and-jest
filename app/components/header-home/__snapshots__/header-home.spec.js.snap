// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View>
    <RCTImageView
      internal_analyticTag={null}
      resizeMode="contain"
      source={
        Array [
          Object {
            "height": undefined,
            "uri": undefined,
            "width": undefined,
          },
        ]
      }
      style={
        Object {
          "height": 40,
          "marginLeft": 8,
          "overflow": "hidden",
          "resizeMode": "contain",
          "width": 40,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginRight": 12,
      }
    }
  >
    <View>
      <RCTImageView
        internal_analyticTag={null}
        resizeMode="contain"
        source={
          Array [
            Object {
              "height": undefined,
              "uri": undefined,
              "width": undefined,
            },
          ]
        }
        style={
          Object {
            "height": 23,
            "overflow": "hidden",
            "resizeMode": "contain",
            "width": 23,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": undefined,
          "justifyContent": "center",
          "width": 16,
        }
      }
    />
    <View>
      <RCTImageView
        internal_analyticTag={null}
        resizeMode="contain"
        source={
          Array [
            Object {
              "height": undefined,
              "uri": undefined,
              "width": undefined,
            },
          ]
        }
        style={
          Object {
            "height": 25,
            "overflow": "hidden",
            "resizeMode": "contain",
            "width": 25,
          }
        }
      />
    </View>
  </View>
</View>
`;
